File: MAStaff-Punishments/Punishments-API/src/main/java/es/angelillo15/mast/api/punishments/cmd/PunishCommand.java
package es.angelillo15.mast.api.punishments.cmd;

import es.angelillo15.mast.api.cmd.Command;
import es.angelillo15.mast.api.cmd.sender.CommandSender;
import es.angelillo15.mast.api.punishments.IPunishPlayer;
import es.angelillo15.mast.api.punishments.PunishPlayersManager;

public abstract class PunishCommand extends Command {
    @Override
    public void onCommand(CommandSender sender, String label, String[] args) {
        if (!(sender.hasPermission("mast.punishments"))) return;

        IPunishPlayer punishPlayer = PunishPlayersManager.getPlayer(sender.getUniqueId());

        if (punishPlayer == null) {
            return;
        }

        onCommand(punishPlayer, label, args);
    }

    public abstract void onCommand(IPunishPlayer sender, String label, String[] args);
}

File: MAStaff-Punishments/Punishments-API/src/main/java/es/angelillo15/mast/api/punishments/cmd/PunishCommand.kt
package es.angelillo15.mast.api.punishments.cmd

import es.angelillo15.mast.api.cmd.Command
import es.angelillo15.mast.api.cmd.sender.CommandSender
import es.angelillo15.mast.api.punishments.IPunishPlayer
import es.angelillo15.mast.api.punishments.PunishPlayersManager
import es.angelillo15.mast.api.thread.execute

abstract class PunishCommand : Command() {
    override fun onCommand(sender: CommandSender, label: String, args: Array<String>) {
        if (!sender.hasPermission("mast.punishments")) return

        val punishPlayer = PunishPlayersManager.getPlayer(sender.uniqueId) ?: return
        execute { onCommand(punishPlayer, label, args) }
    }

    abstract fun onCommand(sender: IPunishPlayer, label: String, args: Array<String>)
}
